convert2rhel/unit_tests/main_test.py

```diff
@@ -28,12 +28,82 @@
 
 from convert2rhel import actions, applock, backup, cert, checks, grub
 from convert2rhel import logger as logger_module
-from convert2rhel import main, pkghandler, pkgmanager, redhatrelease, repo, subscription, toolopts, utils
+from convert2rhel import main, pkghandler, pkgmanager, redhatrelease, repo, subscription, toolopts, unit_tests, utils
 from convert2rhel.actions import report
 from convert2rhel.breadcrumbs import breadcrumbs
 from convert2rhel.systeminfo import system_info
 
 
+class TestRollbackChanges:
+    @pytest.fixture(autouse=True)
+    def mock_rollback_functions(self, monkeypatch):
+        monkeypatch.setattr(backup.changed_pkgs_control, "restore_pkgs", mock.Mock())
+        monkeypatch.setattr(repo, "restore_yum_repos", mock.Mock())
+        monkeypatch.setattr(repo, "restore_varsdir", mock.Mock())
+        monkeypatch.setattr(pkghandler.versionlock_file, "restore", mock.Mock())
+
+    def test_rollback_changes(self, monkeypatch):
+        monkeypatch.setattr(backup.backup_control, "pop_all", mock.Mock())
+
+        main.rollback_changes()
+
+        assert backup.changed_pkgs_control.restore_pkgs.call_count == 1
+        assert repo.restore_yum_repos.call_count == 1
+        assert pkghandler.versionlock_file.restore.call_count == 1
+        # Note: when we remove the BackupController partition hack, the first
+        # of these calls will go away
+        assert backup.backup_control.pop_all.call_args_list == [mock.call(_honor_partitions=True), mock.call()]
+        assert repo.restore_varsdir.call_count == 1
+
+    # The tests below are for the 1.4 hack that splits restore of Changes
+    # managed by the BackupController into two parts: one that executes before
+    # the unported restore items and a second that executes after.  They can be
+    # removed once all the Changes are managed by BackupControl and the
+    # partition code is removed.
```

Comment:
Could you mark what is the last test that will be removed? And if possible, if there is a jira issue would be nice to add as well.

Reply:
Now marked with a comment.